maisim/maisim.Game/Scores/Score.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using maisim.Game.Beatmaps;
+
+namespace maisim.Game.Scores
+{
+    /// <summary>
+    /// Class include the structure of the score object.
+    /// </summary>
+    public class Score
+    {
+        private readonly Beatmap beatmap;
+        private readonly int tap;
+        private readonly int hold;
+        private readonly int slide;
+        private readonly int touch;
+        private readonly float accuracy;
+        private readonly int combo;
+        private readonly int seasonalScore;
+
+        public Score(int tap, int hold, int slide, int touch, float accuracy, int combo, int seasonalScore)
+        {
+            this.tap = tap;
+            this.hold = hold;
+            this.slide = slide;
+            this.touch = touch;
+            this.accuracy = accuracy;
+            this.combo = combo;
+            this.seasonalScore = seasonalScore;
+        }
+
+        public Beatmap Beatmap => beatmap;
+
+        public int Tap => tap;
+
+        public int Hold => hold;
+
+        public int Slide => slide;
+
+        public int Touch => touch;
+
+        public float Accuracy => accuracy;
+
+        public int Combo => combo;
+
+        public int SeasonalScore => seasonalScore;
+
+        public ScoreRank CalculateRank()
```

Comment:
score rank calculation should be moved to a `ScoreProcessor` class and the `Score` itself should store the achieved rank in a property.

Reply:
Still not sure on the `ScoreProcessor` implementation so I implement as the blank class that not get any parameter and use method to get the accuracy, calculate and return `ScoreRank`. Maybe the other way is get the `Score` object as parameter but the bad way of this is you cannot put the `ScoreProcessor` to process the rank and store the `ScoreRank` as the variable in `Score` object.